File: csharp/resistor-color/ResistorColor.cs
using System;
using System.Linq;

public static class ResistorColor
{
    private static readonly string[] ColorNames =
    [
        "black",
        "brown",
        "red",
        "orange",
        "yellow",
        "green",
        "blue",
        "violet",
        "grey",
        "white",
    ];

    public static int ColorCode(string color) => Array.IndexOf(ColorNames, color);

    public static string[] Colors() => ColorNames;
}

File: c/resistor-color/resistor_color.c
#include "resistor_color.h"

#include <stdlib.h>

int color_code(resistor_band_t resistor) { return resistor; }

resistor_band_t *colors() {
  static resistor_band_t colors[WHITE + 1];
  for (resistor_band_t color = BLACK; color <= WHITE; color++) {
    colors[color] = color;
  }
  return colors;
}
